reference-implementation/lib/helpers.js

```diff
@@ -62,6 +62,10 @@ exports.IsFiniteNonNegativeNumber = v => {
 };
 
 exports.InvokeOrNoop = (O, P, args) => {
+  assert(exports.typeIsObject(O) === true);
+  assert(typeof P === 'string');
+  assert(Array.isArray(args));
```

Comment:
Given that the impetus for writing this PR was an accidental bug where `args` was not passed, when it was intended to be passed, I think it hurts, yeah.